torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -42,6 +42,12 @@ enum class OpType : std::uint8_t {
   UNKNOWN = 100,
 };
 
+const std::string GLOO_BACKEND_NAME = "gloo";
+const std::string NCCL_BACKEND_NAME = "nccl";
+const std::string MPI_BACKEND_NAME = "mpi";
+const std::string ROUND_ROBIN_BACKEND_NAME = "round_robin";
+const std::string UNDEFINED = "undefined";
```

Comment:
nit: Probably putting implementation specific constants in an interface file is not right. Can we just use these in the appropriate files for the implementing class?